x-pack/plugins/cases/public/components/attachments/attachments_table.tsx

```diff
@@ -0,0 +1,189 @@
+              <EuiButtonIcon
+                iconType={'download'}
+                aria-label={'download'}
+                href={filesClient.getDownloadHref({
+                  fileKind: CASES_FILE_KINDS[APP_ID].id,
+                  id: attachment.id,
+                })}
+              />
+            );
+          },
+          'data-test-subj': 'attachments-table-action-download',
+        },
+        {
+          name: 'Delete',
+          isPrimary: true,
+          description: 'Delete this file',
+          color: 'danger',
+          icon: 'trash',
+          type: 'icon',
+          onClick: () => {},
+          'data-test-subj': 'attachments-table-action-delete',
+        },
+      ],
+    },
+  ];
+
+  const resultsCount = useMemo(
+    () => (
+      <>
+        <strong>
+          {pagination.pageSize * pagination.pageIndex + 1}
+          {'-'}
+          {pagination.pageSize * pagination.pageIndex + pagination.pageSize}
+        </strong>{' '}
+        {'of'} <strong>{pagination.totalItemCount}</strong>
+      </>
+    ),
+    [pagination.pageIndex, pagination.pageSize, pagination.totalItemCount]
+  );
+
+  return isLoading ? (
+    <EuiLoadingContent data-test-subj="attachments-table-loading" lines={10} />
+  ) : (
+    <>
+      {pagination.totalItemCount > 0 && (
+        <>
+          <EuiSpacer size="xl" />
+          <EuiText size="xs" data-test-subj="attachments-table-results-count">
+            {'Showing'} {resultsCount}
+          </EuiText>
+        </>
+      )}
+      <EuiSpacer size="s" />
+      <EuiBasicTable
+        tableCaption="Attachments Table"
+        items={items}
+        columns={columns}
+        pagination={pagination}
+        onChange={onChange}
+        data-test-subj="attachments-table"
+        noItemsMessage={
+          <EuiEmptyPrompt
+            title={<h3>{'No attachments available'}</h3>}
+            data-test-subj="attachments-table-empty"
+            titleSize="xs"
+            actions={
+              <EuiButton
+                size="s"
+                iconType="plusInCircle"
+                data-test-subj="case-detail-attachments-table-upload-file"
+              >
+                {'Add File'}
+              </EuiButton>
+            }
+          />
+        }
+      />
+      {isModalVisible && (
+        <EuiModal onClose={closeModal}>
+          <EuiModalHeader>
+            <EuiModalHeaderTitle>{selectedFile?.name}</EuiModalHeaderTitle>
+          </EuiModalHeader>
+          <EuiModalBody>
+            <FileImage
+              size="l"
+              alt=""
+              src={filesClient.getDownloadHref({
+                id: selectedFile?.id || '',
+                fileKind: CASES_FILE_KINDS[APP_ID].id,
+              })}
+            />
+          </EuiModalBody>
+          <EuiModalFooter>
+            <EuiButton size="s" onClick={closeModal} fill>
+              {'Close'}
+            </EuiButton>
+          </EuiModalFooter>
+        </EuiModal>
+      )}
```

Comment:
I think is better to move the modal logic to another component (`FilePreviewer`). We would need it in the user activity.